File: perl/get-files.pl
# Let's just guess at stuff until it works.

use strict;
use warnings;

use File::Find;

my $max_indent_level = 2;
my %ignored_paths = map { $_ => 1 } ('.', '..', '.git', 'node_modules', 'target', 'love', 'test_app', 'zef');

sub list_files {
    return if exists($ignored_paths{$_});
    print("$File::Find::name\n");
}

find(\&list_files, ('..'))


# Local Variables:
# compile-command: "perl ./get-files.pl"
# End:

    

File: cpp/get-files.cpp
#include <string>
#include <iostream>
#include <filesystem>
namespace fs = std::filesystem;

int main() {
  std::string const DIRECTORY_PATH = "..";
  std::string ignored_paths[4] = {".git", "love", "target", "node_modules"};
  
  for (auto& entry : fs::recursive_directory_iterator(DIRECTORY_PATH)) {
    bool should_skip = false;
    for (auto ignored_path : ignored_paths) {
      if (entry.path().string().find(ignored_path) != std::string::npos)
        should_skip = true;
    }
    if (should_skip) continue;
    std::cout << entry.path().lexically_relative(DIRECTORY_PATH) << std::endl;
  }
}

// Local Variables:
// compile-command: "clang++ -Wall -std=c++17 ./get-files.cpp -lstdc++fs -o get-files && ./get-files"
// End:
